src/core_editor/line_buffer.rs

```diff
@@ -479,36 +470,24 @@ mod test {
     #[test]
     fn set_buffer_updates_insertion_point_to_new_buffer_length() {
         let mut line_buffer = buffer_with("test string");
-        let before_operation_location = InsertionPoint {
-            line: 0,
-            offset: 11,
-        };
+        let before_operation_location = InsertionPoint { offset: 11 };
         assert_eq!(before_operation_location, line_buffer.insertion_point());
 
         line_buffer.set_buffer("new string".to_string());
 
-        let after_operation_location = InsertionPoint {
-            line: 0,
-            offset: 10,
-        };
+        let after_operation_location = InsertionPoint { offset: 10 };
         assert_eq!(after_operation_location, line_buffer.insertion_point());
     }
 
     #[test]
     fn set_buffer_works_with_multi_line_string() {
         let mut line_buffer = buffer_with("test string");
-        let before_operation_location = InsertionPoint {
-            line: 0,
-            offset: 11,
-        };
+        let before_operation_location = InsertionPoint { offset: 11 };
         assert_eq!(before_operation_location, line_buffer.insertion_point());
 
         line_buffer.set_buffer("new line 1\nnew_line 2".to_string());
 
-        let after_operation_location = InsertionPoint {
-            line: 1,
-            offset: 10,
-        };
+        let after_operation_location = InsertionPoint { offset: 10 };
```

Comment:
This does not seem right. The old test was saying that the line is first and the location on that line is 10. Maybe let's assert for the line being 1 here too. Thoughts?

Reply:
Ignore this one, looks like the last commit fixes this.